lib/ui/events.js

```diff
@@ -0,0 +1,317 @@
+'use strict';
+
+const blessed     = require('blessed'),
+      moment      = require('moment'),
+      { spinner } = require('./ui'),
+      util        = require('../util'),
+      yaml        = require('json2yaml');
+
+
+class Events {
+  constructor ({ screen, client, debug, cancellations, key, navbar, namespace }) {
+    const { until } = spinner(screen);
+    // TODO: Add filtering
+    const events_table = blessed.listtable({
+      label         : key,
+      parent        : screen,
+      top           : 1,
+      width         : '100%',
+      height        : '50%-1',
+      border        : 'line',
+      align         : 'left',
+      keys          : true,
+      tags          : true,
+      mouse         : true,
+      noCellBorders : true,
+      scrollbar     : {
+        ch    : ' ',
+        style : { bg: 'white' },
+        track : {
+          style : { bg: 'grey' },
+        }
+      },
+      style : {
+        label  : { bold: true },
+        border : { fg: 'white' },
+        header : { fg: 'grey' },
+        cell   : { fg: 'white', selected: { bg: 'blue' } },
+      }
+    });
+    const event_description = blessed.box({
+      label         : 'Event description',
+      parent        : screen,
+      top           : '50%',
+      bottom        : '1',
+      width         : '100%',
+      height        : '50%-1',
+      border        : 'line',
+      align         : 'left',
+      keys          : true,
+      tags          : true,
+      mouse         : true,
+      noCellBorders : true,
+      scrollable    : true,
+      scrollbar     : {
+        ch    : ' ',
+        style : { bg: 'white' },
+        track : {
+          style : { bg: 'grey' },
+        }
+      },
+      style : {
+        label  : { bold: true },
+        border : { fg: 'white' },
+        header : { fg: 'grey' },
+        cell   : { fg: 'white', selected: { bg: 'blue' } },
+      }
+    });
+
+
+    // do not render when detached. 
+    events_table._orender = events_table.render;
+    events_table.render = function () {
+      if (this.parent == null) {
+        return;
+      }  
+      this._orender();
+    };
+
+    // work around to preserve selection and scrolling
+    events_table.on('detach', () => {
+      const { selected, childBase, childOffset } = events_table;
+      events_table.once('attach', () => {
+        events_table.select(selected);
+        Object.assign(events_table, { childBase, childOffset });
+        events_table.scrollTo(selected);
+      });
+    });
+
+    events_table.on('key C-z', () => {
+      cancellations.run(key);
+      navbar.remove(key);
+    });
+
+    const { promise, cancellation } = client.events(namespace).get({ cancellable : true });
+    cancellations.add(key, cancellation);
+    until(promise)
+    .spin(s => events_table.setLabel(`${s} ${key}`))
```

Comment:
I'd suggest to follow current pattern used to label widgets, e.g., `Events [namespace]`.